r/src/compute.cpp

```diff
@@ -449,7 +449,7 @@ std::shared_ptr<arrow::compute::FunctionOptions> make_compute_options(
     return std::make_shared<Options>(cpp11::as_cpp<std::string>(options["characters"]));
   }
 
-  if (func_name == "utf8_slice_codeunits") {
+  if (func_name == "utf8_slice_codeunits" || func_name == "binary_slice") {
```

Comment:
@rok @AlenkaF  Would one of you have to time to help @kshitij12345 write some R test here?

Reply:
It's ok for me if done on a separate PR.